tests/lean/run/utf8英語.lean

```diff
@@ -1,3 +1,47 @@
+import Lean.Util.TestExtern
+
+instance : BEq ByteArray where
+  beq x y := x.data == y.data
+
+test_extern String.toUTF8 ""
+test_extern String.toUTF8 "\x00"
+test_extern String.toUTF8 "$£€𐍈"
+
+macro "test_extern'" t:term " => " v:term : command =>
+  `(test_extern $t
+    #guard $t == $v)
```

Comment:
Should we have this next to `test_extern` with a more helpful name? Maybe
`guard_extern`? Or simply use `test_extern` and look for the `=>` syntax to enable the additional guard?